src/Utilities/ToolTask.cs

```diff
@@ -1365,7 +1365,24 @@ public override bool Execute()
                     }
                     else
                     {
-                        File.AppendAllText(_temporaryBatchFile, commandLineCommands, EncodingUtilities.CurrentSystemOemEncoding);
+
+                        Encoding encoding;
+
+                        if (Traits.Instance.EscapeHatches.AvoidUnicodeWhenWritingToolTaskBatch)
+                        {
+                            encoding = EncodingUtilities.CurrentSystemOemEncoding;
+                        }
+                        else
+                        {
+                            encoding = EncodingUtilities.BatchFileEncoding(commandLineCommands + _temporaryBatchFile, EncodingUtilities.UseUtf8Detect);
+
+                            if (encoding.CodePage != EncodingUtilities.CurrentSystemOemEncoding.CodePage)
+                            {
+                                File.AppendAllText(_temporaryBatchFile, $@"%SystemRoot%\System32\chcp.com {encoding.CodePage}>nul{Environment.NewLine}"); // first line is ANSI
```

Comment:
First line is decoded with the console code page which will default to the OEM code page (may be distinct from ANSI, though all chars here should be ASCII).

Reply:
Clarified this comment, thanks.